### DISCUSSION TITLE
Renaming Blocks

### DISCUSSION THREAD

**Discussion by TekyTeky** [Novakasa/brickrail#139] [General]
I see that any new Blocks can be named, which is great, is it possible to rename existing Blocks?

**Comment by Novakasa** [comment]
Right now, unfortunately not, I have tried to do it but the way the name is distributed across internal state currently it would make it very complicated and likely lead to bugs. I do have a plan of refactoring internal state so the name is not used as internal reference anymore, and then I will easily be able to make the names editable. This is tracked in https://github.com/Novakasa/brickrail/issues/122

For now, a workaround is to make a backup of your layout file, open the layout.brl in a text editor, and replace all instances of "old-blockname" with "new-blockname" (make sure you also include the quotes in the replacement, so you don't accidentally replace part of "block0__" when renaming "block0_".

**Reply by TekyTeky** [reply]
Ok, thanks.. I can do the workaround easily!!

**Comment by Novakasa** [comment]
BTW, this is now possible as well with #122 mostly completed. In the train and block inspector, you can click on the name to rename them.

**Reply by TekyTeky** [reply]
Thanks so much!
